borsh/src/ser/mod.rs

```diff
@@ -236,6 +291,43 @@ impl<T> BorshSerialize for HashSet<T>
     }
 }
 
+impl<K, V> BorshSerialize for BTreeMap<K, V>
+where
+    K: BorshSerialize + PartialOrd,
+    V: BorshSerialize,
+{
+    #[inline]
+    fn serialize<W: Write>(&self, writer: &mut W) -> Result<()> {
+        let mut vec = self.iter().collect::<Vec<_>>();
+        vec.sort_by(|(a, _), (b, _)| a.partial_cmp(b).unwrap());
```

Comment:
This sort seems unnecessary since https://doc.rust-lang.org/std/collections/struct.BTreeMap.html#method.iter says to return as sorted

Reply:
Great catch! I will fix it